tests/app/storage/test_redis.py

```diff
@@ -1,214 +1,173 @@
-            self.redis.get(EQSession, "test")
-
-        # Then
-        assert self.redis.client.get.call_count == 2
-
-    def test_delete_handles_connection_error_once(self):
-        # Given
-        used_at = datetime.now(tz=timezone.utc)
-        expires_at = used_at + timedelta(seconds=60)
-        jti = UsedJtiClaim(str(uuid.uuid4()), expires_at)
-
-        self.redis.client.delete = mock.Mock(
-            side_effect=[RedisConnectionError, RedisConnectionError]
-        )
-
-        # When
-        with self.assertRaises(RedisConnectionError):
-            self.redis.delete(jti)
-
-        # Then
-        assert self.redis.client.delete.call_count == 2
+def test_put_jti_stores_empty_value(redis, redis_client):
+    used_at = datetime.now(tz=timezone.utc)
+    expires_at = used_at + timedelta(seconds=60)
+
+    jti = UsedJtiClaim(str(uuid.uuid4()), expires_at)
+
+    redis.put(jti, overwrite=False)
+
+    stored_data = redis_client.get(jti.jti_claim)
+
+    assert b"" == stored_data
+
+
+def test_duplicate_put_jti_fails(redis):
+    used_at = datetime.now(tz=timezone.utc)
+    expires_at = used_at + timedelta(seconds=60)
+
+    jti = UsedJtiClaim(str(uuid.uuid4()), expires_at)
+
+    redis.put(jti, overwrite=False)
+
+    with pytest.raises(ItemAlreadyExistsError):
+        redis.put(jti, overwrite=False)
+
+
+def test_put_session(redis, eq_session):
+    # given
+
+    stored_data = redis.get(EQSession, eq_session.eq_session_id)
+    assert stored_data is None
+
+    # when
+    redis.put(eq_session)
+
+    # Then
+    stored_data = redis.get(EQSession, eq_session.eq_session_id)
+    assert stored_data is not None
+
+
+def test_get_session(redis, eq_session):
+    # Given
+    stored_data = redis.get(EQSession, eq_session.eq_session_id)
+    assert stored_data is None
+    redis.put(eq_session)
+
+    # When
+    stored_data = redis.get(EQSession, eq_session.eq_session_id)
+
+    # Then
+    for k, v in eq_session.__dict__.items():
+        parsed_value = getattr(stored_data, k)
+        if isinstance(v, datetime):
+            assert v >= parsed_value
+        else:
+            assert v == parsed_value
+
+
+def test_delete_session(redis, eq_session):
+    # Given
+    redis.put(eq_session)
+    session = redis.get(EQSession, "sessionid")
+    assert session.eq_session_id == eq_session.eq_session_id
+
+    # When
+    redis.delete(eq_session)
+
+    # Then
+    assert redis.get(EQSession, "sessionid") is None
+
+
+def test_redis_does_not_store_key_field_in_value(redis, redis_client, eq_session):
+    # Given
+    stored_data = redis.get(EQSession, eq_session.eq_session_id)
+    assert stored_data is None
+    redis.put(eq_session)
+
+    # When
+    stored_data = redis_client.get(eq_session.eq_session_id)
+
+    storage_model = StorageModel(model_type=EQSession)
+
+    assert storage_model.key_field not in json_loads(stored_data)
+
+
+def test_get_redis_expiry_when_expiry_set(redis, redis_client, eq_session):
+    # Given
+    # When
```

Comment:
No Given?

Reply:
Yeah the Given was just the existence of a session https://github.com/ONSdigital/eq-questionnaire-runner/blob/master/tests/app/storage/test_redis.py#L128-L132 which is now the `eq_session` fixture